### ISSUE TITLE
Operate e2e protococal fails at "Map names to UUID"

### ISSUE THREAD

**Issue by rodrigo-lourenco-lopes** [camunda/zeebe#14780]
**Describe the bug**

In the process e2e protocol we hava one of the branches failing to retrive it's generation in the task "Map names to UUID" with the following error message:

```
java.lang.IllegalArgumentException: Unable to find generation '3d8f8033-1d1c-42f5-bc83-994e8feddf33'; closest candidates: ['4a80d8a3-c5e2-4d20-8653-504acb14eb73', 'ad079f43-3f83-42e3-aabc-e99cd7de569c', '4495a935-b0e4-4e05-bc36-fd19e455e4f4']; available candidates: ['1bede723-f469-456f-9449-ae6b7a064ae0', '760e9f07-a0e5-4742-8a86-07d7243d0e47', 'dce9bbfa-06c2-4563-a3af-b3c7d3999fe9', '6d7ad8f0-5948-4404-9383-87399a554110', '4a80d8a3-c5e2-4d20-8653-504acb14eb73', 'a941b647-4084-4d7f-be60-83f08e469bbb', '6cfd80be-f32f-4291-885a-63e6673e4441', '7d0ab39a-e957-4457-8302-5fb97e556869', 'f32dd0e6-8ef7-4855-b715-ae454a32c5e0', '1f9213f5-7781-4fc1-859d-4639411e8ffa', '7b4a61f8-55f5-4219-abd3-f4a554749dbc', 'acbb44eb-0923-4b2a-83a9-151937b2f7df', 'bc05a2ca-f670-4ab5-acf9-f7ab7f400b9d', 'cfdcfc5d-8c99-49be-a396-6464d5fab645', '06ca3dd9-8d63-4b59-bf30-05e5f384938e', '74ab4eea-ba18-49d8-9066-e02753fd7664', '6e16431e-b49f-4df7-9abc-6adc5a82a898', '30edad6e-85b3-4fa0-9357-1ff6993db4bf', '50c15215-efa5-41a7-9a0f-8463720acc35', 'c0a2373d-8e24-4509-a845-3a75bdca438a', '94b299e7-f602-4a3e-baf0-20cd867917ea', '6a37b6e0-a5bf-4ffb-9cb4-5a104bad7c44', 'dc615859-98b9-48ff-9f6c-046f3c129e9e', '9f6ab595-bfbb-4a13-8bd3-84eebdee0d3e', '170e5a39-1901-4e52-87eb-8e6657e4d1fe', 'ad079f43-3f83-42e3-aabc-e99cd7de569c', '88be82cc-0f17-4c1f-884a-cd0681f8232e', '07ccfc44-0c41-4318-b7f5-65c2afc37d7c', '07aa03f8-c53b-4650-80ca-81865480a442', '3a5309b9-5890-49d6-af2e-d88627ffe123', 'aa25b1b0-6ea8-4eb1-9b4e-ce95aef7ead6', '4495a935-b0e4-4e05-bc36-fd19e455e4f4', 'df53a5c3-a6bc-4180-b062-d1b9d977eb1d', '7c53bc8f-5634-4b7b-9b53-8bd3c4a60db0', '7b0d03d0-811c-437a-878a-67db73940605', '8e64a853-f346-4620-bb8b-ed931e820011', '3693d923-2344-48d7-acd3-f1b1d31d0662', '42778581-baf1-4ed6-9e33-b8cf39ee30d6', '0e6d5a7e-8e8e-481a-bb78-bd9dc5da4054', 'c52b7f02-e284-4956-800f-e48f179568d9', '9b077d71-cc1b-4503-82b7-2090f70329b1', 'cc5c11e3-e7ab-4312-9a46-df4ba050525c', 'aef17377-ff9b-4734-bad6-5e71cd96b990', '244f93bf-ef1c-4314-ae91-30dfa6543da3', '4c4a19c2-3412-4770-8380-2c39c1e51559', '02bd3686-aea4-4b23-bbb0-47c4e1c97812', '6d568596-0d2a-4e6e-9b2d-0929a98ee50a', 'f39cedf0-6da8-40f3-aa90-0cf7fbb6de8b', '4d1b8c40-6ad8-427f-acd7-bcb66bbe9869', '9437a247-2386-49f5-842c-bfaec2b3b13e', '091023e2-1944-4ccb-ae42-7ba031d1562d', 'ba34bd1a-8521-4059-847f-37a50a06789c', 'a86363d2-9cf0-44e9-b004-97ea15d509bd', '67992ea5-41b3-4eb2-9db0-65830941fd75', 'f1e7bf99-3514-44c0-8cc1-a7587779f810', '86a6fa57-1f6a-4c92-b445-fa713e26a6e0', '40e1f202-af3f-45a9-8773-a75172e44c36', '42ee3cac-4aff-4992-bfc7-e162ae7fa71a', 'd93c624d-49ae-473f-837b-724fe3fb33fe', '308c42fb-ad4e-4d46-90d4-16768a5edb82', 'fef937bd-f4c4-4c6b-896f-cfdf16694e35', 'd6242ef7-5a86-45b1-90de-da9a35fa78a6', '7c3eb2e9-d3b2-4194-b0f3-af61401a7ede', '2e9a7859-198d-4f85-92f0-826674827884', 'b4027591-144a-410a-9b5c-d6a2d9a3978c', 'a6270b18-66fe-4a26-81de-7269459a291b', '2593dc8e-734b-4e83-8eb5-f1aeb4fdc7d6', '19638238-fb2d-4780-82b4-3c5515034676', '1f550983-5c02-4fdc-96cb-7ffb5e46d8bd', 'a0b58424-353a-41d3-8e06-54eeeb5d9739', 'fd1094fb-37d4-48cb-bbd3-b7517e6adacb', '0af5dec2-1c5f-4465-8393-a27212ead646', '20455d44-e09f-4b6e-9c85-a2df0de86915']
	at io.zeebe.clustertestbench.handler.MapNamesToUUIDsHandler.lambda$mapGeneration$1(MapNamesToUUIDsHandler.java:109)
	at io.vavr.control.Either.getOrElseThrow(Either.java:342)
	at io.zeebe.clustertestbench.handler.MapNamesToUUIDsHandler.mapGeneration(MapNamesToUUIDsHandler.java:109)
	at io.zeebe.clustertestbench.handler.MapNamesToUUIDsHandler.handle(MapNamesToUUIDsHandler.java:39)
	at io.zeebe.clustertestbench.handler.JobHandlerWithEnrichedLogger.handle(JobHandlerWithEnrichedLogger.java:29)
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.executeJob(JobRunnableFactory.java:44)
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.lambda$create$0(JobRunnableFactory.java:39)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```

Process variables:

Name | Value
-- | --
branch | "stable/8.1"
build | "https://github.com/camunda/zeebe/actions/runs/6530122147"
channel | "Internal Dev"
clusterPlan | "Production - M"
generation | "temp_Ij837wingj"
generationTemplate | "Camunda 8.1.18"
generationUUID | "3d8f8033-1d1c-42f5-bc83-994e8feddf33"
properties | ["allInstancesAreCompleted"]
region | "Chaos, Belgium, Europe (europe-west1)"
testParams | {"faultDuration":"PT1M","verifier":{"maxInstanceDuration":"15m"},"faultRecoveryWaitTime":"PT1M","maxTestDuration":"P5D","loopWaitDuration":"PT5M","fault":null,"starter":[{"rate":20,"processId":"one-task-one-timer"},{"rate":10,"processId":"ping-pong-message"}]}
testProcessId | "e2e-test"
zeebeImage | "gcr.io/zeebe-io/zeebe:8.1.18-SNAPSHOT-stable-8.1-ce2abbdf"

**To Reproduce**

**Expected behavior**

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
<STACKTRACE>
```

</p>
</details>

**Environment:**
- OS: 
- Zeebe Version: 
- Configuration:

**Comment by npepinpe** [comment]
Triage: 

As this directly affects medic duties, and I'm medic next week, I'll try to work on it during the week.

TODO: move to testbench repository

**Comment by Zelldon** [comment]
@npepinpe we can pair on this when I'm back hopefully next week🤞I think I already have good knowledge about this

See also related https://camunda.slack.com/archives/CKZK2E7RP/p1697542535727689?thread_ts=1697542535.727689&cid=CKZK2E7RP

This week I also added more output to testbench to see which endpoints are called.

**Comment by Zelldon** [comment]
First this is likely a duplicate of https://github.com/zeebe-io/zeebe-cluster-testbench/issues/941. 

I will close the other one, since it was opened later and contains less information. I will. transfer this issue to testbench.

**Comment by Zelldon** [comment]
Interesting. I can't transfer the issue to another org. Ok then we keep it here.

**Comment by Zelldon** [comment]
## Investigation

_Up front, I wanted to share that I agreed with Schuppi that we take a closer look and investigate more before we trigger in ask-console again and again. Because it doesn't look like that the issue is on their side, at least we need to prove it first._


**What we know:** 

 * The E2E tests seem to fail with this error almost every week, at least some of them.
 * The problem seems to be that the channel doesn't get updated in time and we have to manually update it before we can continue.
 * Interesting is that the normal QA tests don't run into this issue. 
 * We have recently migrated the testbench to use the external API for the cloning of generations.
 * At first we thought this is related to that (but this happened already before and this was the reason why we migrated.)
 

I'm wondering how the E2E and QA processes differ and whether they have something as variables defined that it might to cause to fail these requests 🤔 I will check the processes and also the inputs of the process instances. 

I expect that both use the same service task but I will check this as well.

**Comment by Zelldon** [comment]
## Investigation Pt. 2

**TL;DR;** We used in E2E a different generation template. A generation that wasn't part of the Internal Channel, which is used by Testbench to create clusters. 

----

I had a look at the process models and the process instance variables. It turned out that the variables differ in the generation template

**QA Process execution:**
![testbench-generation-vars](https://github.com/camunda/zeebe/assets/2758593/47066d91-3ad3-47d6-9da2-c08986fd1279)


**E2E process execution:**
![e2e-generation-vars](https://github.com/camunda/zeebe/assets/2758593/1fc82d8c-a791-4630-95bf-3b9ad1d99aec)

We can see that E2E is using a more specific generation, which might be the most recently released. In one case the version was not even available in SaaS (v8.1.20). 

To understand what the underlying issue here is we need to first clarify how the [Clone endpoint from Console works](https://console.cloud.dev.ultrawombat.com/external/openapi/docs/#/default/CreateClone).

When using the new clone endpoint of the Console API we use the generation template as a reference to pre-fill all details (from the template) this includes versions but also the channels it is part of. We can then specify overwrites like the zeebe docker image for the to be created generation.

All of these generations which were referenced by the E2E process instances were not part of the Internal Channel. This means when creating/cloning the referenced generation and setting the right Zeebe image everything works as expected (from console side), but from out side not. Testbench expects to create a cluster with a new generation which is part of the Internal Channel, this is the reason why we see the following errors all the time 
🔲 


![error](https://github.com/camunda/zeebe/assets/2758593/696266f6-a3df-492a-a4ca-7612b8373268)

## Fixes and follow-up

To fix this we should either default back to the snapshot generation (as this is per default in the Internal Channel), or make sure that we always have the referenced generation in the expected channel.

We probably should also add a new service task which should check this up-front, to make sure that all our invariants or expectations are true.
